src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -354,7 +354,12 @@ class DefaultSource
       dataRepartitioned.foreachPartition((rows: Iterator[Row]) => {
         import CdpConnector.ioRuntime
 
-        val maxParallelism = Math.max(1, config.partitions / numberOfPartitions)
+        val maxParallelism = if (config.maxWritePartitions.isDefined) {
+          1
+        } else {
+          Math.max(1, config.partitions / numberOfPartitions)
+        }
+
```

Comment:
New idea:
- Remove everything related to `maxWritePartitions` everywhere. Both here and in jetfire.
- In [TransformService](https://github.com/cognitedata/jetfire-backend/blob/8ae98aa94d46b0cbc3bb3e1181888e5c837f3785/driver/src/main/scala/com/cognite/jetfire/driver/TransformService.scala#L295), I repartition if the destination is WDL.
- Manually set the `partitions` property when writing to WDL. This is necessary, otherwise, it will try to run many batches in parallel even if the number of spark partitions is low.